aggregator/src/aggregation/rlc/gates.rs

```diff
@@ -0,0 +1,427 @@
+use halo2_proofs::{
+    circuit::{AssignedCell, Cell, Region, RegionIndex, Value},
+    halo2curves::bn256::Fr,
+    plonk::Error,
+};
+use zkevm_circuits::util::Challenges;
+
+use crate::{constants::LOG_DEGREE, util::assert_equal};
+
+use super::RlcConfig;
+
+impl RlcConfig {
+    /// initialize the chip with fixed 0 and 1 cells
+    pub(crate) fn init(&self, region: &mut Region<Fr>) -> Result<(), Error> {
+        region.assign_fixed(|| "const zero", self.fixed, 0, || Value::known(Fr::zero()))?;
+        region.assign_fixed(|| "const one", self.fixed, 1, || Value::known(Fr::one()))?;
+        region.assign_fixed(|| "const two", self.fixed, 2, || Value::known(Fr::from(2)))?;
+        region.assign_fixed(|| "const four", self.fixed, 3, || Value::known(Fr::from(4)))?;
+        region.assign_fixed(
+            || "const eight",
+            self.fixed,
+            4,
+            || Value::known(Fr::from(8)),
+        )?;
+        region.assign_fixed(
+            || "const thirty two",
+            self.fixed,
+            5,
+            || Value::known(Fr::from(32)),
+        )?;
+        Ok(())
+    }
+
+    #[inline]
+    pub(crate) fn zero_cell(&self, region_index: RegionIndex) -> Cell {
+        Cell {
+            region_index,
+            row_offset: 0,
+            column: self.fixed.into(),
+        }
+    }
+
+    #[inline]
+    pub(crate) fn one_cell(&self, region_index: RegionIndex) -> Cell {
+        Cell {
+            region_index,
+            row_offset: 1,
+            column: self.fixed.into(),
+        }
+    }
+
+    #[inline]
+    pub(crate) fn two_cell(&self, region_index: RegionIndex) -> Cell {
+        Cell {
+            region_index,
+            row_offset: 2,
+            column: self.fixed.into(),
```

Comment:
These functions are not safe in my opinion. If we store these fixed cells in RlcConfig object, then we can just return them here.

Reply:
fixed with 47beb9e969cd82033d4f7449ecd8a02e7003a1b0

Reply:
tried a few method and did not work. reverting back to the old version.

PS: For this circuit, the rlc_config only have one region.